capital/code.ts

```diff
@@ -0,0 +1,50 @@
+// This plugin fetches a list of countries from an external API and shows the capital of the selected country
+
+let countries = [];
+
+// Create an invisible iframe UI to use network API's
+figma.showUI(
+  `<script>
+    (async (event) => {
+      const res = await fetch("https://api.sampleapis.com/countries/countries");
+      const json = await res.json();
+      window.parent.postMessage({ pluginMessage: json }, "*");
+    })();
+  </script>`,
+  { visible: false }
+);
+
+figma.parameters.on('input', ({ key, query, result }: ParameterInputEvent) => {
+  // When fetching data from an external source, it is recommended to show a relevant loading message
+  result.setLoadingMessage('Loading countries...');
+
+  // Set the suggestions after the a message has been received from the iframe
+  figma.ui.onmessage = (json) => {
+    countries = json.map((country, index) => {
+      return {
+        name: country.name,
+        data: {
+          index: index,
+          name: country.name,
+          capital: country.capital,
+        },
+      };
+    });
+    result.setSuggestions(countries);
+  };
```

Comment:
Pretty minor, but I don't think this needs to be in the figma.parameters.on() method. Perhaps move it outside just to show that they aren't actually related?

Reply:
I put it in the method to have access to `result`. Can you show me how I can take it out and call `result.setSuggestions`?

Reply:
Good idea! I've moved it out.